File: LC1854.py
class Solution:
    def maximumPopulation(self, logs: List[List[int]]) -> int:
        yr = [0]*2051

        for i in logs:
            yr[i[0]] += 1
            yr[i[1]] -= 1

        cnt = yr[1950]; early = 1950

        for i in range(1951, 2051):
            yr[i] += yr[i-1]

            if yr[i] > cnt:
                cnt = yr[i]
                early = i
        
        return early

File: LC1854.cpp
class Solution {
public:
    int maximumPopulation(vector<vector<int>>& logs) {
        vector<int> yr (2051, 0);

        for(auto i : logs){
            yr[i[0]]++;
            yr[i[1]]--;
        }
        int cnt = yr[1950], early = 1950;

        for(int i = 1951; i < 2051; i++){
            yr[i] += yr[i-1];           // Prefix sum

            if(yr[i] > cnt){
                cnt = yr[i];
                early = i;
            }
        }

        return early;
    }
};
